src/App.tsx

```diff
@@ -1,27 +1,65 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
-import { GoodsList } from './GoodsList';
+import { GoodsList } from './components/GoodsList/GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
+import { Button } from './components/Button';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+const defaultGoods: Good[] = [];
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState(defaultGoods);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState('');
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const handleGoodsLoad = async (callback: () => Promise<Good[]>) => {
```

Comment:
Every time I press the same button it makes a new request. Add state for the last clicked button and check it before loading.

Reply:
Buttons are disabled while loading 
![image](https://user-images.githubusercontent.com/11507341/226972365-eff3f9d3-0374-4370-bdcd-8b75ddad3730.png)

Reply:
Try it yourself. Look Network tab in dev tools.